activestorage/lib/active_storage/service/s3_service.rb

```diff
@@ -12,9 +12,9 @@ module ActiveStorage
   # See ActiveStorage::Service for the generic API documentation that applies to all services.
   class Service::S3Service < Service
     attr_reader :client, :bucket
-    attr_reader :multipart_upload_threshold, :upload_options
+    attr_reader :multipart_upload_threshold, :upload_options, :download_options
 
-    def initialize(bucket:, upload: {}, public: false, **options)
+    def initialize(bucket:, upload: {}, public: false, download: {}, **options)
       @client = Aws::S3::Resource.new(**options)
```

Comment:
`download` key is not reserved for S3 client (https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#initialize-instance_method).